### PR TITLE
Complete minimal implementation

### PR DESCRIPTION
**Pull request by Sadere**

### CODE REVIEW

**Review by arttet — CHANGES_REQUESTED** [review]

File: proto/keeper/v1/secrets.proto [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+message UploadFileV1Request {
+  string metadata = 1;
+  string file_name = 2;
+  string master_password = 3;
+  bytes chunk = 4;
+}
+
+message DownloadFileV1Request {
```
[the remark this answers is not in the document]
Comment (Sadere) [reply proto/keeper/v1/secrets.proto]: fixed

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.22
+FROM golang:1.22.4
```
Comment (arttet) [inline docker/Dockerfile:1]: You need to use multi-stages build.
https://docs.docker.com/build/building/multi-stage/

File: internal/client/api/client.go [hunk 3]
```diff
@@ -1,10 +1,24 @@
+// Provides interface for API communication
 package api
 
 import (
 	"context"
+
+	"github.com/Sadere/gophkeeper/pkg/model"
 )
 
+//go:generate mockgen -source client.go -destination mocks/mock_client.go -package api
```
Comment (arttet) [inline internal/client/api/client.go:10]: It would be better if it was in the `mockgen` package.

File: internal/client/api/grpc/notification.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+		// Trigger secret list reload
+		if p != nil {
+			p.Send(msg.ReloadSecretList{})
+		}
+	}
+}
+
+func (c *GRPCClient) sleep() {
+	time.Sleep(time.Second * 2)
```
Comment (arttet) [inline internal/client/api/grpc/notification.go:59]: This is a magic number

**Review by arttet — APPROVED** [review]
Good job!

**Pull request merged**